plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php

```diff
@@ -340,6 +352,108 @@ function ( $subscription ) {
 		}
 	}
 
+	/**
+	 * Runs on in_plugin_update_message-{file-name}, show a message if plugin is without a subscription.
+	 * Only Woo local plugins are passed to this function.
+	 *
+	 * @see setup_message_for_plugins_without_subscription
+	 * @param object $plugin_data An array of plugin metadata.
+	 * @param object $response  An object of metadata about the available plugin update.
+	 *
+	 * @return void.
+	 */
+	public static function display_notice_for_plugins_without_subscription( $plugin_data, $response ) {
+		// Extract product ID from the response.
+		$product_id = preg_replace( '/[^0-9]/', '', $response->id );
+
+		if ( WC_Helper::has_product_subscription( $product_id ) ) {
+			return;
+		}
+
+		// Prepare the expiry notice based on subscription status.
+		$purchase_link = add_query_arg(
+			array(
+				'utm_source'   => 'pu',
+				'utm_campaign' => 'pu_plugin_screen_purchase',
+			),
+			PluginsHelper::WOO_SUBSCRIPTION_PAGE_URL
+		);
+
+		$product_price = self::get_regular_price_for_product_without_subscription( $product_id );
+		/* translators: 1: Product regular price */
+		$product_price_formatted = $product_price ? sprintf( __( ' for %s ', 'woocommerce' ), $product_price ) : '';
+
+		$notice = sprintf(
+			/* translators: 1: URL to My Subscriptions page 2: Product price */
+			__( ' You don\'t have a subscription, <a href="%1$s" class="woocommerce-purchase-subscription">subscribe%2$s</a> to update.', 'woocommerce' ),
+			esc_url( $purchase_link ),
+			$product_price_formatted
+		);
+
+		// Display the expiry notice.
+		echo wp_kses(
+			$notice,
+			array(
+				'a' => array(
+					'href'  => array(),
+					'class' => array(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Gets the price of a product without a subscription as that is not present in the update data.
+	 * This uses the WooCommerce.com API to get the price of the product.
+	 *
+	 * @param mixed $product_id The product ID.
+	 * @return null|string The price of the product.
+	 */
+	public static function get_regular_price_for_product_without_subscription( $product_id ) {
```

Comment:
@vishnugopal Since the price is not available directly, and we need to make an extra API call, does it make sense to get rid of the price info from this alert? So the notices would be without the ` for %s` and everything else would be same.  

We were hoping to minimize the number of external calls (cc: @gedex)

Reply:
@mehamasum  I can do that as well if that's the design decision we're taking. I have tried to follow best practices and added caching and timeouts for this API call, but will defer to your guidance here.

Reply:
I agree with removing extra remote request just for the price info.

Reply:
Removed in da6854304f.